addons/web/static/src/views/fields/section_note_list/section_note_list_field.js

```diff
@@ -0,0 +1,387 @@
+            const moveAfterRecord = (
+                isSubSection
+                    ? item.records
+                    : [item.record, ...item.childItems.map((it) => it.record)]
+            ).at(-1);
+            const context = isSubSection ? this.makeSectionCreationContext(true) : {};
+            const newRecord = await this.props.list._createNewRecordDatapoint({
+                context,
+                manuallyAdded: true,
+                mode: "edit",
+            });
+            await this.props.list._addRecord(newRecord, {});
+            await this.props.list._resequence(newRecord.id, moveAfterRecord.id);
+            await this.props.list._onUpdate();
+        });
+    }
+
+    buildHierarchy(list) {
+        const itemMap = {};
+        const hierarchy = new HierarchyItem(DISPLAY_TYPES.ROOT, {}, null);
+        let parent = hierarchy;
+
+        for (const record of list.records) {
+            const type = record.data[this.props.displayTypeField] || DISPLAY_TYPES.RECORD;
+            if (isSection(type)) {
+                if (type === DISPLAY_TYPES.SECTION) {
+                    parent = hierarchy;
+                } else if (parent.type === DISPLAY_TYPES.SUBSECTION) {
+                    parent = parent.parent;
+                }
+            }
+            const item = new HierarchyItem(type, record, parent);
+            parent.addChild(item);
+            itemMap[item.id] = item;
+            if (item.isSection) {
+                parent = item;
+            }
+        }
+
+        return { hierarchy, itemMap };
+    }
+
+    /**
+     * @param {HierarchyItem} item
+     */
+    canAddSubSection(item) {
+        const displayTypeField = this.props.list.fields[this.props.displayTypeField];
+        const selection = new Map(displayTypeField.selection);
+        return selection.has(DISPLAY_TYPES.SUBSECTION) && item.isTopSection;
+    }
+
+    /**
+     * @param {HierarchyItem} item
+     */
+    async deleteSection(item) {
+        if (this.editedRecord && this.editedRecord !== item) {
+            const left = await this.props.list.leaveEditMode();
+            if (!left) {
+                return;
+            }
+        }
+        if (this.activeActions.onDelete) {
+            const method = this.props.isM2M ? "unlink" : "delete";
+            const commands = item.records.map((r) =>
+                x2ManyCommands[method](r.resId || r._virtualId)
+            );
+            await this.props.list.model.mutex.exec(async () => {
+                await this.props.list._applyCommands(commands);
+                await this.props.list._onUpdate();
+            });
+        }
+    }
+
+    /**
+     * @param {HierarchyItem} item
+     */
+    async duplicateSection(item) {
+        await this.props.list.model.mutex.exec(async () => {
+            const promises = [];
+            for (const record of item.records) {
+                const data = { ...record.data };
```

Comment:
this part isn't correct. `record.data` contains StaticList datapoints for x2many fields, and you can't share them to another datapoint. Also, for many2one fields it's an object, I'm not a fan either to give the same reference.

I think you should manually generate create commands here, and then call `applyCommands` (new function suggested above)